005-tailwindcss-framework.md

```diff
@@ -0,0 +1,38 @@
+
+**Discussion:** <https://github.com/DoSomething/rfcs/pull/5>
+
+## Problem
+
+It has become more difficult over time for our tech team to confidently update the styling and design patterns of our front-end user interfaces.
+
+While our pattern library, [Forge](https://github.com/DoSomething/forge), worked great for building interfaces with the established patterns it provided, it has not been updated in some time and with our rapidly changing design needs and team structure, it no longer seems like the ideal solution to continue evolving.
+
+As our team of full-stack developers shifts to doing more work on our projects that includes both back-end and front-end changes, we need a styling framework in place that is flexible, customizable and empowers our developers to confidently use without the need to code custom style declarations themselves.
+
+Additionally, the patterns in Forge are much more specific to how the DoSomething.org website looked in prior iterations. As the design evolved, it has forced us to override many of Forge's rules to be able to replicate changing design needs.
+
+## Proposal
+
+By utilizing the Tailwind CSS Framework the tech team can work more confidently when building the front-end for different interfaces.
+
+A utility class framework like Tailwind CSS offers the following benefits:
+
+- We can easily style items and pages by applying a pre-configured library of classes directly to HTML.
+- We can style custom components without needing to write custom CSS, and without needing to invent specific and unique class names for each component.
+- Once we establish our collection of utility classes, a project's CSS stops growing, since there is less need for custom CSS.
+- Helps keeps CSS scoped to a component, resulting in less worries that a style change could affect styles globally.
+- We will be building interfaces with proper constraints defined by the framework's configuration; no more magic pixel value numbers.
+- The utilities would help developers adhere to the predefined design system, keeping our interfaces consistent.
+- Support for responsive design and pseudo-classes within the series of generated utility classes.
+- If needed, we can extract components when it makes sense.
+
+## Forge 7.0
+
+We think that for the next version of Forge, it could embrace the utility class approach and still be the go-to package for styling front-end interfaces on DoSomething applications.
+
+For version 7.0 we propose eliminating all the current patterns in Forge (as our projects switch to using Tailwind, they will stop using these patterns as is and thus no longer needed). Instead Forge would house the main TailwindCSS configuration and provide the generated styles from this custom configuration. Projects that use Forge 7.0 package can include these generated TailwindCSS styles and have access to all the utility classes they otherwise need.
```

Comment:
What would this mean for much of the styling on Phoenix where we rely on current Forge patterns -- would we just need to do a clean sweep of those classes in favor of the Tailwind classes before updating to `7.0`? (Though how would we do that _without_ having Tailwind installed from 7.0 in the first place?)

Reply:
I think the process would be:

1. Implement Tailwind in Phoenix.
2. Proceed with updating Phoenix to utilize Tailwind classes instead of Forge or custom ones (so Forge no longer plays a role).
3. Update Forge to 7.0 which will consist of the Tailwind config (pulled from Phoenix) and any components deemed reusable across platforms.
4. Re-implement Forge (v7.0) into Phoenix and remove the Tailwind config and reusable components from Phoenix since now Forge v7.0 will provide them.

I _think_ that's the general gist.